pkg/assembler/backends/testing/backend.go

```diff
@@ -134,3 +208,117 @@ func filterQualifiersAndSubpath(v *model.PackageVersion, pkgSpec *model.PkgSpec)
 	}
 	return v
 }
+
+func filterSourceNamespace(src *model.Source, sourceSpec *model.SourceSpec) (*model.Source, error) {
+	var namespaces []*model.SourceNamespace
+	for _, ns := range src.Namespaces {
+		if sourceSpec.Namespace == nil || ns.Namespace == *sourceSpec.Namespace {
+			newNs, err := filterSourceName(ns, sourceSpec)
+			if err != nil {
+				return nil, err
+			}
+			if newNs != nil {
+				namespaces = append(namespaces, newNs)
+			}
+		}
+	}
+	if len(namespaces) == 0 {
+		return nil, nil
+	}
+	return &model.Source{
+		Type:       src.Type,
+		Namespaces: namespaces,
+	}, nil
+}
+
+func filterSourceName(ns *model.SourceNamespace, sourceSpec *model.SourceSpec) (*model.SourceNamespace, error) {
+
+	var names []*model.SourceName
+	for _, n := range ns.Names {
+		if sourceSpec.Name == nil || n.Name == *sourceSpec.Name {
+			n, err := filterQualifier(n, sourceSpec)
```

Comment:
Hmm it looks like we've implemented "no checking for qualifiers" differently here from https://github.com/guacsec/guac/pull/397, I think both ways are fine, but probably want to be consistent..

Thoughts @mihaimaruseac ?

Reply:
gotcha - yea i think this is probably something that needs to be documented with more nuance (i opened #401 )